In [1]:
from IPython.display import display
import numpy as np

<p style="font-size: 30px; line-height: 135%;}">
<b>Calidad de un vino</b>
</p>

<p style="font-size: 25px; line-height: 135%;}">
<b>Pregunta a</b>
</p>

In [2]:
import pandas as pd
df_red = pd.read_csv("winequality-red.csv",sep=";")
df_white = pd.read_csv("winequality-white.csv",sep=";")
df = pd.concat([df_red,df_white], axis=0)

df = df.assign(Tinto = pd.Series(np.array([1] * len(df_red) + [0] * len(df_white))).values)

#display(df)
display(df.describe())
df.info()

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    6497.000000       6497.000000  6497.000000     6497.000000   
mean        7.215307          0.339666     0.318633        5.443235   
std         1.296434          0.164636     0.145318        4.757804   
min         3.800000          0.080000     0.000000        0.600000   
25%         6.400000          0.230000     0.250000        1.800000   
50%         7.000000          0.290000     0.310000        3.000000   
75%         7.700000          0.400000     0.390000        8.100000   
max        15.900000          1.580000     1.660000       65.800000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  6497.000000          6497.000000           6497.000000  6497.000000   
mean      0.056034            30.525319            115.744574     0.994697   
std       0.035034            17.749400             56.521855     0.002999   
min       0.009000             1.000000              6.000000     0.987110   
25%       0.038000            17.000000             77.000000     0.992340   
50%       0.047000            29.000000            118.000000     0.994890   
75%       0.065000            41.000000            156.000000     0.996990   
max       0.611000           289.000000            440.000000     1.038980   

                pH    sulphates      alcohol      quality        Tinto  
count  6497.000000  6497.000000  6497.000000  6497.000000  6497.000000  
mean      3.218501     0.531268    10.491801     5.818378     0.246114  
std       0.160787     0.148806     1.192712     0.873255     0.430779  
min       2.720000     0.220000     8.000000     3.000000     0.000000  
25%       3.110000     0.430000     9.500000     5.000000     0.000000  
50%       3.210000     0.510000    10.300000     6.000000     0.000000  
75%       3.320000     0.600000    11.300000     6.000000     0.000000  
max       4.010000     2.000000    14.900000     9.000000     1.000000

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6497 entries, 0 to 4897
Data columns (total 13 columns):
fixed acidity           6497 non-null float64
volatile acidity        6497 non-null float64
citric acid             6497 non-null float64
residual sugar          6497 non-null float64
chlorides               6497 non-null float64
free sulfur dioxide     6497 non-null float64
total sulfur dioxide    6497 non-null float64
density                 6497 non-null float64
pH                      6497 non-null float64
sulphates               6497 non-null float64
alcohol                 6497 non-null float64
quality                 6497 non-null int64
Tinto                   6497 non-null int64
dtypes: float64(11), int64(2)
memory usage: 710.6 KB


<p style="font-size: 18px; line-height: 135%;}">
<b>a)</b> El dataset presenta un total de 6497 registros, estos con 13 atributos no nulos. Se pueden ver estadisticos relacionados a cada atributo en la tabla mostrada con el método ``df.describe()``
</p>

<p style="font-size: 25px; line-height: 135%;}">
<b>Pregunta b</b>
</p>

In [3]:
df['good_quality'] = [1 if q>5 else 0 for q in df.quality]
del df['quality']

In [4]:
from sklearn.model_selection import train_test_split

atributos = list(df)
atributos.remove("good_quality")

X = df.as_matrix(atributos)
y = df.as_matrix(["good_quality"])

#Se proceden a generar los conjuntos
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=1)
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, test_size=0.25, random_state=1)

<p style="font-size: 25px; line-height: 135%;}">
<b>Pregunta c</b>
</p>